### ISSUE TITLE
Avoid using "PUT" or "DELETE" HTTP methods

### ISSUE THREAD

**Issue by DanReyLop** [woocommerce/woocommerce-admin#1934]
Originally discussed in https://github.com/woocommerce/woocommerce/issues/23155

The WooCommerce REST API uses all the HTTP verbs (`GET`, `POST`, `PUT`, `DELETE`). Also, we've found (props @dechov for the find) that WooCommerce Admin is already relying on some `PUT` REST endpoints. [Here](https://github.com/woocommerce/woocommerce-admin/blob/4c6e02a927332515272c5585c6e506d551d2088c/client/wc-api/notes/operations.js#L84), for example.

In WooCommerce Services, we've had user reports in the past regarding endpoints like those. An `apache2` server, with `mod_security` configured with the most popular set of settings, will forbid any `PUT` or `DELETE` HTTP requests to hit the server at all.

Here's the issue, with more context: https://github.com/Automattic/woocommerce-services/issues/930

Gutenberg fixed the same issue here: https://github.com/WordPress/gutenberg/pull/4396 and here: https://github.com/WordPress/gutenberg/pull/5741

Basically, the fix is to change all `PUT` requests that com from the client to `POST` requests with a `X-HTTP-Method-Override: PUT` HTTP header, or a `_method=PUT` querystring arg. Same with `DELETE` or `PATCH` requests.

**Comment by DanReyLop** [comment]
Upon further investigation, you're using `@wordpress/api-fetch` in the client, which ends up including this middleware: https://github.com/WordPress/gutenberg/blob/d9768ad4294ffc08bed68459d342fa746f63aea4/packages/api-fetch/src/middlewares/http-v1.js

So, by virtue of using Gutenberg, this issue doesn't apply to you. Sorry for the noise :)
